src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+
+    } else if (item == ITEM_STICK_UPGRADE_30) {
+        if (INV_CONTENT(ITEM_STICK) != ITEM_STICK) {
+            INV_CONTENT(ITEM_STICK) = ITEM_STICK;
+        }
+        Inventory_ChangeUpgrade(UPG_STICKS, 3);
+        AMMO(ITEM_STICK) = CAPACITY(UPG_STICKS, 3);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_NUT_UPGRADE_30) {
+        if (INV_CONTENT(ITEM_NUT) != ITEM_NUT) {
+            INV_CONTENT(ITEM_NUT) = ITEM_NUT;
+        }
+        Inventory_ChangeUpgrade(UPG_NUTS, 2);
+        AMMO(ITEM_NUT) = CAPACITY(UPG_NUTS, 2);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_NUT_UPGRADE_40) {
+        if (INV_CONTENT(ITEM_NUT) != ITEM_NUT) {
+            INV_CONTENT(ITEM_NUT) = ITEM_NUT;
+        }
+        Inventory_ChangeUpgrade(UPG_NUTS, 3);
+        AMMO(ITEM_NUT) = CAPACITY(UPG_NUTS, 3);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_STICK) {
+        if (INV_CONTENT(ITEM_STICK) != ITEM_STICK) {
+            Inventory_ChangeUpgrade(UPG_STICKS, 1);
+            AMMO(ITEM_STICK) = 1;
+        } else {
+            AMMO(ITEM_STICK)++;
+            if (AMMO(ITEM_STICK) > CUR_CAPACITY(UPG_STICKS)) {
+                AMMO(ITEM_STICK) = CUR_CAPACITY(UPG_STICKS);
+            }
+        }
+    } else if ((item == ITEM_STICKS_5) || (item == ITEM_STICKS_10)) {
+        if (INV_CONTENT(ITEM_STICK) != ITEM_STICK) {
+            Inventory_ChangeUpgrade(UPG_STICKS, 1);
+            AMMO(ITEM_STICK) = sAmmoRefillCounts[item - ITEM_STICKS_5];
+        } else {
+            AMMO(ITEM_STICK) += sAmmoRefillCounts[item - ITEM_STICKS_5];
+            if (AMMO(ITEM_STICK) > CUR_CAPACITY(UPG_STICKS)) {
+                AMMO(ITEM_STICK) = CUR_CAPACITY(UPG_STICKS);
+            }
+        }
+
+        item = ITEM_STICK;
+
+    } else if (item == ITEM_NUT) {
+        if (INV_CONTENT(ITEM_NUT) != ITEM_NUT) {
+            Inventory_ChangeUpgrade(UPG_NUTS, 1);
+            AMMO(ITEM_NUT) = 1;
+        } else {
+            AMMO(ITEM_NUT)++;
+            if (AMMO(ITEM_NUT) > CUR_CAPACITY(UPG_NUTS)) {
+                AMMO(ITEM_NUT) = CUR_CAPACITY(UPG_NUTS);
+            }
+        }
+
+    } else if ((item == ITEM_NUTS_5) || (item == ITEM_NUTS_10)) {
+        if (INV_CONTENT(ITEM_NUT) != ITEM_NUT) {
+            Inventory_ChangeUpgrade(UPG_NUTS, 1);
+            AMMO(ITEM_NUT) += sAmmoRefillCounts[item - ITEM_NUTS_5];
+        } else {
+            AMMO(ITEM_NUT) += sAmmoRefillCounts[item - ITEM_NUTS_5];
+            if (AMMO(ITEM_NUT) > CUR_CAPACITY(UPG_NUTS)) {
+                AMMO(ITEM_NUT) = CUR_CAPACITY(UPG_NUTS);
+            }
+        }
+        item = ITEM_NUT;
+
+    } else if (item == ITEM_POWDER_KEG) {
+        if (INV_CONTENT(ITEM_POWDER_KEG) != ITEM_POWDER_KEG) {
+            INV_CONTENT(ITEM_POWDER_KEG) = ITEM_POWDER_KEG;
+        }
+
+        AMMO(ITEM_POWDER_KEG) = 1;
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BOMB) {
+        if ((AMMO(ITEM_BOMB) += 1) > CUR_CAPACITY(UPG_BOMB_BAG)) {
```

Comment:
Does it have to be in the if?

Reply:
Unfortunately yes. This part was terrible to figure out the solution

Reply:
Urgh.

Reply:
Does it work as `AMMO(ITEM_BOMB)++`?

Reply:
Nope, it actually doesn't lol. It has to be `+= 1`.
`++` breaks matching

Reply:
They aren't the same, right? `x++` returns `x`, whereas `x += 1` returns `x + 1`. So it's either
```c
x++
if (x > y)
```
or
```c
if (++x > y)
```